src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -1735,43 +1735,37 @@ private void GrowTable(Tables tables)
                     return;
                 }
 
-
                 // Compute the new table size. We find the smallest integer larger than twice the previous table size, and not divisible by
                 // 2,3,5 or 7. We can consider a different table-sizing policy in the future.
-                int newLength = 0;
-                bool maximizeTableSize = false;
+                int newLength = MaxArrayLength;
                 try
                 {
                     checked
                     {
                         // Double the size of the buckets table and add one, so that we have an odd integer.
-                        newLength = tables._buckets.Length * 2 + 1;
+                        int newLengthTmp = tables._buckets.Length * 2 + 1;
 
                         // Now, we only need to check odd integers, and find the first that is not divisible
                         // by 3, 5 or 7.
-                        while (newLength % 3 == 0 || newLength % 5 == 0 || newLength % 7 == 0)
+                        while (newLengthTmp % 3 == 0 || newLengthTmp % 5 == 0 || newLengthTmp % 7 == 0)
                         {
-                            newLength += 2;
+                            newLengthTmp += 2;
                         }
 
-                        Debug.Assert(newLength % 2 != 0);
+                        newLength = newLengthTmp;
 
-                        if (newLength > MaxArrayLength)
-                        {
-                            maximizeTableSize = true;
-                        }
+                        Debug.Assert(newLengthTmp % 2 != 0);
                     }
                 }
                 catch (OverflowException)
                 {
-                    maximizeTableSize = true;
                 }
 
-                if (maximizeTableSize)
+                if (newLength >= MaxArrayLength)
```

Comment:
It looks like this condition would have only hit if newLength > MaxArrayLength, but now it's >=?

Reply:
Yes, that's because when the OverflowException triggers `newLength` won't have been set and it will still hold its initial value `MaxArrayLength`. However, we still want to set `_budget = int.MaxValue;`.